### ISSUE TITLE
Cancel index background worker when exiting JabRef

### ISSUE THREAD

**Issue by ColinScarato** [JabRef/jabref#8468]
Jabref 5.5 automatically looks for files to autolink with all entries when opening a database. I use a relatively big database whose linked files are stored on a network drive; every time I open it Jabref tries to freshly scan to autolink files, as it probably believes that the files in the network drive might have changed.
This results in 10 minutes of background tasks running, meaning that every time I open and close Jabref in less than a 10 minutes I get an additional prompt to wait for background tasks to finish running.

Would it be possible to add a setting to disable file autolinking?
Thanks!

**Comment by tobiasdiez** [comment]
Do you really mean the autolink or fulltext indexing? I noticed this problem with the latter but not the former.

I think it would already help with the background task would just be canceled when exiting. There is no need to finish the indexing if you close the app anyway ;-)

cc @btut

**Comment by ColinScarato** [comment]
![autolink](https://user-images.githubusercontent.com/55202645/151408599-4288f5cf-dc60-4b27-9ed2-6db9743417bb.png)
That looks like indexing :)

**Comment by tobiasdiez** [comment]
While the option to completely disable index is certainly very nice to have (thanks @btut!), the issue that the index background worker is not cancel automatically when JabRef is closed still remains.

**Comment by ColinScarato** [comment]
I'm not sure why it fails to recognise indexed files, it doesn't show me more info. I just saw that it does it correctly locally, but not on that shared drive.
Disabling indexing is fine to me personally. Given that we don't have a completely streamlined pdf organisation yet, and that auto-indexing seems to fail for network shares anyway, we link the pdfs manually.

**Comment by ColinScarato** [comment]
Yes I tried to let it run (on several PCs). I'm not sure if I can see logs of what happened.

**Comment by btut** [comment]
Thanks. I don't think there would be logs as I don't have any idea why this could be failing. Could you share:
- what operating system you use
- what kind of network share you are using
- how you mount the network share

Then I could investigate what the issue may be.

**Comment by tobiasdiez** [comment]
For me it seems to just iterate the files, without actually indexing it. But even this takes about a minute for 2k entries. I could imagine that on a shared network service it is even slower.

> So you want to have an option to disable the graceful-close dialog that asks to quit while tasks are running? Why would you want that?

No, I just want that it exists immediately without any user interaction, and then picks up the reindexing once I start JabRef again. I don't think indexing is such an important task that needs to be finished before I'm allowed to close JabRef.
